social-network-api/src/main/java/fs/socialnetworkapi/service/PostService.java

```diff
@@ -57,18 +57,20 @@ public List<PostDtoOut> getAllPost(Integer page, Integer size) {
 
   public List<PostDtoOut> getProfilePosts(Integer page, Integer size) {
 
-    User user = getUser();
+    User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
Why won't you use getUser() method? If the logic is changed, it would be worse to rewrite it in all class methods.